### ISSUE TITLE
Add the __ow_headers to the ow object

### ISSUE THREAD

**Issue by kptdobe** [adobe/openwhisk-action-utils#28]
The CoralogixLogger and the PapertrailLogger automatically add an `ow` object to the log message: 

```
ow: {
          activationId: process.env.__OW_ACTIVATION_ID,
          actionName: process.env.__OW_ACTION_NAME,
          transactionId: process.env.__OW_TRANSACTION_ID
        },
```

This is really useful and this information could be enhanced with a `header` property containing the value of the incoming `params.__ow_headers`.
This would be required to achieve: https://github.com/adobe/helix-home/issues/70

Based on @tripodsan feedback, the full `ow` object (+ the new `headers`) should be computed directly in the generic helix-log logger. (details https://github.com/adobe/openwhisk-action-utils/commit/63a529110992ab694983ab4eab1def25b1c0dbf8#commitcomment-35805854) 

I started to prototype this but I hit 2 road blockers:
- I do not really understand the logger structure, the final `__ow_logger` object contains a nested structure of 4 level of MultiLoggers. This should be fixed with #26 
- There is no generic `sanitize` method to prevent the headers to contain secrets. See #27

**Comment by tripodsan** [comment]
the headers should not be logged with every log message, IMO. just with the first one. that's why I suggested to place them into the `wrap` code.

also, the `__ow_logger` is a bunyan logger as written in the jsdoc. not a multilogger.

**Comment by kptdobe** [comment]
Correct, but it contains a steam with a BunyanStreamInterface with a steam with a MultiLogger which finally contains the ConsoleLogger and the CoralogixLogger. That's when I got confused on how to move the formatter defined in the CoralogixLogger to a higher level and make sure all entries get the headers.

Having all the headers on all entries is the only way to get all logs for on "request" in one shot. If we log them only for the first one, we have to extract the activationid and start more complex queries to get all logs in one shot.

**Comment by tripodsan** [comment]
> Having all the headers on all entries is the only way to get all logs for on "request" in one shot. If we log them only for the first one, we have to extract the activationid and start more complex queries to get all logs in one shot.

still, the amount of headers is insane. if you transfer them for every log entry to coralogix, our bill will explode.... I would only include the header that you need to identify the log.

**Comment by tripodsan** [comment]
> Correct, but it contains a steam with a BunyanStreamInterface with a steam with a MultiLogger which finally contains the ConsoleLogger and the CoralogixLogger. That's when I got confused on how to move the formatter defined in the CoralogixLogger to a higher level and make sure all entries get the headers.
> 
> Having all the headers on all entries is the only way to get all logs for on "request" in one shot. If we log them only for the first one, we have to extract the activationid and start more complex queries to get all logs in one shot.

yes. now that we have the new log-fields support in helix-log, we can probably make this simpler...I will look into it.

**Comment by kptdobe** [comment]
We could also:
- log only the headers the first time
- each time, use my initial idea and compute a `referrer` entry based on the `x-forwarded-host`, `x-forwarded-port`, `x-forwarded-proto`, `x-old-url` headers. Like this we have the necessary info with only few extra characters.

**Comment by kptdobe** [comment]
Back to ground 0: for whatever reason, the `x-forwarded-host` header value changed today and is now always `adobeioruntime.net` which makes it useless. 
Will be back to my initial idea of adding an extra `x-referrer` header.

**Comment by trieloff** [comment]
> the `x-forwarded-host` header value changed today and is now always adobeioruntime.net which makes it useless.

Can we check with the Adobe I/O Runtime team first?

**Comment by kptdobe** [comment]
I asked on the slack channel, let see what they have to say.
